### DISCUSSION TITLE
Preparing database relationships for Directus mirror

### DISCUSSION THREAD

**Discussion by Thermal-Shock** [directus/directus#12148] [Archive]
We are currently evaluating Directus and I was wondering if there was any documentation or guide that describes how to have various o2m, m2m, m2o etc relationships mirrored  correctly by Directus as their respective relationships?

The database we have does not currently have all the foreign keys that it should and also has tables with multiple primary keys for things like language translations. As part of the migration to Directus, we need to modify our schema and code base so they are correctly prepared and mirrored smoothly into Directus.

For example, let's say I have an orders tables that has a created by user id and a users table. So we have a m2o relationship. Is it enough to simply have foreign keys for directus to convert the orders.user_id field into a relationship? 

I wouldn't worry necessarily, but it doesn't look like there is a way to change a field to a relationship without removing it and recreating it, thereby losing the data? This could get difficult to correct after the mirroring interpretation when multiple relationships are involved. 

For our tables that have multiple primary keys, we most likely need to create a junction table, but again does this have to be done in very specific way in order for Directus to mirror everything correctly? 

I will do some experiments by creating each relationship type in directus and look at the resulting tables/relationships, so if I go through that exercise and apply that to the schema I want to mirror, will directus recognize it along with the correct field types?

All the best.

**Comment by rijkvanzanten** [comment]
Directus will indeed rely on foreign key constraints to extract what relationships exist. By adding a foreign key constraint to your m2o fields, Directus should correctly pick them up as relational fields 👍🏻 

As for composite/compound primary keys, Directus currently doesn't support that, so you might run into some issues with those particular tables
